src/mongo/db/auth/oidc/oidc_identity_providers_registry.h

```diff
@@ -0,0 +1,126 @@
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    Server Side Public License for more details.
+
+    You should have received a copy of the Server Side Public License
+    along with this program. If not, see
+    <http://www.mongodb.com/licensing/server-side-public-license>.
+
+    As a special exception, the copyright holders give permission to link the
+    code of portions of this program with the OpenSSL library under certain
+    conditions as described in each individual source file and distribute
+    linked combinations including the program with the OpenSSL library. You
+    must comply with the Server Side Public License in all respects for
+    all of the code used other than as permitted herein. If you modify file(s)
+    with this exception, you may extend this exception to your version of the
+    file(s), but you are not obligated to do so. If you do not wish to do so,
+    delete this exception statement from your version. If you delete this
+    exception statement from all source files in the program, then also delete
+    it in the license file.
+======= */
+
+#pragma once
+
+#include "mongo/db/auth/oidc/oidc_server_parameters_gen.h"
+#include "mongo/util/periodic_runner.h"
+#include <boost/optional.hpp>
+#include <memory>
+#include <ranges>
+#include <unordered_map>
+
+namespace mongo {
+
+class ServiceContext;
+class JWKSFetcherFactory;
+namespace crypto {
+class JWKManager;
+}  // namespace crypto
+
+
+// Constructs and registers the OIDC identity providers registry in the given ServiceContext.
+void initializeOidcIdentityProvidersRegistry(ServiceContext* serviceContext);
+
+/**
+ * Holds and manages all configured OIDC identity providers, along with associated
+ * JWK managers and periodic tasks.
+ *
+ * This registry is constructed once during server initialization and remains
+ * immutable for the lifetime of the ServiceContext. The identity provider
+ * configurations, associated JWK managers, and periodic jobs are fixed at
+ * construction time and not meant to be modified at runtime.
+ *
+ * If support for dynamic updates or reloads is required in the future,
+ * appropriate synchronization must be implemented to ensure thread safety.
+ */
+class OidcIdentityProvidersRegistry {
+public:
+    // Returns the registry instance from service context decoration.
+    static OidcIdentityProvidersRegistry& get(ServiceContext* serviceContext);
+
+    // Sets the registry instance in service context decoration.
+    static void set(ServiceContext* serviceContext,
+                    std::unique_ptr<OidcIdentityProvidersRegistry> registry);
+
+    /**
+     * Constructs the OidcIdentityProvidersRegistry with a const lvalue reference to the given
+     * configuration list.
+     *
+     * The registry does not take ownership of the configuration vector and assumes it remains valid
```

Comment:
I chose this approach to avoid copying, but I'm not sure it was the right choice, as improper usage can lead to undefined behavior.